File: Backend/src/shared/services/groq.service.ts
import Groq from "groq-sdk";
import dotenv from 'dotenv';
dotenv.config();

const groq = new Groq({ apiKey: process.env.GROQ_API_KEY });

export async function groqGeneratedData(prompt: string) {
    const chatComplete = groq.chat.completions.create({
        messages: [
            {
                role: "user",
                content: prompt
            }
        ],
        model: "llama-3.3-70b-versatile",
        response_format: { type: "json_object" }
    });

    return (await chatComplete).choices[0]?.message?.content || ""
}

File: Backend/src/services/groq_service.py
from groq import AsyncGroq
from src.core.config import settings

groq_client = AsyncGroq(api_key=settings.GROQ_API_KEY)

async def groq_generated_data(prompt: str) -> str:
    """
    Generate data using the Groq API expecting JSON formatted responses.
    """
    chat_completion = await groq_client.chat.completions.create(
        messages=[
            {
                "role": "user",
                "content": prompt,
            }
        ],
        model="openai/gpt-oss-120b",
        response_format={"type": "json_object"}
    )
    
    return chat_completion.choices[0].message.content or ""
